rca/enquire_to_study/views.py

```diff
@@ -189,14 +189,32 @@ def send_user_email_notification(self, form):
             fail_silently=False,
         )
 
+    def set_session_data(self, form_data, enquiry_submission):
+        # Sets form data into session variable for analytics.
+        self.request.session["enquiry_form_session_data"] = {
+            "event": "form-submission",
+            "formName": "Enquire to study form",
+            "formId": enquiry_submission.id,
+            "countryResidence": form_data["country_of_residence"],
+            "countryCitizen": form_data["country_of_citizenship"],
+            "programme": [i.title for i in form_data["programmes"]],
+            "starting": form_data["start_date"].label,
+            "enquiryType": form_data["enquiry_reason"].reason,
+            "newsletter": True,
+        }
+
     def form_valid(self, form):
         country_of_residence = form.cleaned_data["country_of_residence"]
         if country_of_residence in ["GB", "IE"]:
             self.post_mailchimp()
         else:
             self.post_qs(form.cleaned_data)
 
-        self.create_form_submission(form)
+        enquiry_submission = self.create_form_submission(form)
```

Comment:
I'm not a huge fan of the way this reads, or if it's bad practice. `create_form_submission` just calls `form.save()` so I'm just grabbing what I need by returning the [form.save()](https://github.com/torchbox/rca-wagtail-2019/pull/608/files#diff-52c257a033b79d236a9b23e7a64206ff76ae355021264e863d1083791b1eb889R165)

Reply:
Yeah I'm not sure what benefit `create_form_submission` is, seems like a method just to call another method on the same class?

Reply:
I think in this instance it's turned out handy so I can retrieve the submission object created in form.save()

Reply:
Would `enquiry_submission = form.save()` not work?

Reply:
:facepalm: good idea